src/cli/cli.cpp

```diff
@@ -3263,6 +3263,27 @@ otError Interpreter::ProcessPing(uint8_t aArgsLength, char *aArgs[])
 
     memset(&config, 0, sizeof(config));
 
+    if (aArgsLength >= 2)
+    {
+        if (!strcmp(aArgs[0], "-I"))
+        {
+            SuccessOrExit(error = ParseAsIp6Address(aArgs[1], config.mSource));
+            bool                  valid        = false;
+            const otNetifAddress *unicastAddrs = otIp6GetUnicastAddresses(mInstance);
+            for (const otNetifAddress *addr = unicastAddrs; addr; addr = addr->mNext)
+            {
+                if (otIp6IsAddressEqual(&addr->mAddress, &config.mSource))
+                {
+                    valid = true;
+                    break;
+                }
+            }
+            VerifyOrExit(valid, error = OT_ERROR_INVALID_ARGS);
```

Comment:
Skip valid checking if REFERENCE_DEVICE == 1 ?
I think we may need to test packets with invalid source address.